azext_iot/operations/hub.py

```diff
@@ -1964,13 +1965,13 @@ def _build_device_or_module_connection_string(entity, key_type="primary"):
     )
     auth = entity["authentication"]
     auth_type = auth["type"].lower()
-    if auth_type == "sas":
+    if auth_type == HubDeviceAuthType.sas.name:
         key = "SharedAccessKey={}".format(
             auth["symmetricKey"]["primaryKey"]
             if key_type == "primary"
             else auth["symmetricKey"]["secondaryKey"]
         )
-    elif auth_type in ["certificateauthority", "selfsigned"]:
+    elif auth_type in [HubDeviceAuthType.certificateAuthority.name, HubDeviceAuthType.selfSigned.name]:
```

Comment:
This is a bug, the auth_type is lower cased first.

Reply:
Thanks for the catch! :)